src/InvariantTheory/invariant_rings.jl

```diff
@@ -127,16 +127,81 @@ function Base.show(io::IO, IR::InvRing)
 end
 
 function reynolds_molien_via_singular(IR::InvRing)
+   @assert !ismodular(IR)
    if !isdefined(IR, :reynolds_singular) || !isdefined(IR, :molien_singular)
       R = polynomial_ring(IR)
       singular_matrices = _action_singular(IR)
-      rey, mol = Singular.LibFinvar.reynolds_molien(singular_matrices...)
+
+      if iszero(characteristic(coefficient_ring(IR)))
+         rey, mol = Singular.LibFinvar.reynolds_molien(singular_matrices...)
+      else
+         error("Not implemented yet") # See https://github.com/oscar-system/Singular.jl/issues/486
+      end
       IR.reynolds_singular = rey
       IR.molien_singular = mol
    end
    return IR.reynolds_singular, IR.molien_singular
 end
 
+function reynolds_operator(IR::InvRing{FldT, GrpT, T}, f::T) where {FldT, GrpT, T <: MPolyElem}
+   @assert parent(f) === polynomial_ring(IR)
+
+   rey, _ = reynolds_molien_via_singular(IR)
+   fSing = singular_ring(polynomial_ring(IR))(f)
+   fReySing = Singular.LibFinvar.evaluate_reynolds(rey, fSing)
+   # fReySing is an ideal...
+   @assert length(gens(fReySing)) == 1
+   return polynomial_ring(IR)(gens(fReySing)[1])
+end
+
+function molien_series(IR::InvRing)
+   @assert !ismodular(IR)
+   _, mol = reynolds_molien_via_singular(IR)
```

Comment:
For reference, in GAP you can compute the Molien series of a matrix group in char 0 as follows:
````
gap> G:=Group([ [ [ 0, -1 ], [ 1, -1 ] ] ]);;
gap> psi:=NaturalCharacter(G);;
gap> mol:= MolienSeries( psi );
( 1-z+z^2 ) / ( (1-z^3)*(1-z) )
```
Of course this won't work for non-integer Oscar matrices; but we could start by supporting it there, and then reimplement the underlying algorithm in Julia (I have a reference for it somewhere, need to dig it out)